Replace empty lines with the indentation from the previous line before sending to the REPL

resolritter opened the pull request:
The Python REPL interprets an empty line as the termination of inputs,
so this is needed to tell it we might not be done with the input, e.g.

```python
class Something:
    b = 1

    # python repl will interpret the previous empty line as the end of the class
    # thus the method definition below will error
    def method():
        return self.b
```

I chose to implement it here because other REPLs might do the same

hkupty commented:
I was just going to mention about last line being used to trigger the evaluation. Is this latest patch tested that it works? Can I merge it?

resolritter commented:
@hkupty I moved it to Draft, and back, because this line was problematic

```lua
indent = string.match(v, '^(%s)')
```

The problem is that `string.match` might return `nil` if the line has no indent, thus a `nil` line would be put into the input, which of course isn't valid. I've changed it to

```lua
indent = string.match(v, '^(%s)') or ""
```

i.e. if the previous line doesn't have an indent, it'll fall back to an empty string

hkupty commented:
:+1: Nice! Thanks for the PR! This was much needed for python.
